numpy/_core/function_base.py

```diff
@@ -119,6 +126,11 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
     >>> plt.show()
 
     """
+    if device not in ["cpu", None]:
```

Comment:
I think rather than check `device` here, it should just be passed on to `_nx.arange()`

Reply:
Sure! Removed.